crates/ruff_linter/src/rules/flake8_bugbear/rules/getattr_with_constant.rs

```diff
@@ -75,7 +69,7 @@ pub(crate) fn getattr_with_constant(
     if is_mangled_private(value.to_str()) {
         return;
     }
-    if !checker.semantic().is_builtin("getattr") {
+    if !checker.semantic().match_builtin_expr(func, "getattr") {
```

Comment:
The counterpoint is that we're now doing _other_ work that isn't strictly necessary (for example, if this is a name, and builtins isn't imported, and the name isn't `getattr`, then the call to `is_mangled_private` wasn't necessary -- we could've exited much earlier; similarly, we risk calling `is_identifier` on many more function calls that won't ever match).

Reply:
This is clearly still the right tradeoff, but that's the tradeoff.

Reply:
Correct

Reply:
The only way around this would be to, like, have some `could_be_builtin("getattr")` method you call at the top that returns `false` if builtins isn't imported, it's not a name, or the name isn't `getattr`, but IDK, that seems not great / worthwhile.

Reply:
These things (like `is_identifier`) _do_ show up in traces sometimes because we tend to be calling these rules for _every_ function call, which is why being able to gate on something as cheap as "the name of the function" is useful.

Reply:
I don't think we need to change anything here but wanted to raise visibility on it.